luigi/s3.py

```diff
@@ -77,22 +77,41 @@ def __init__(self, aws_access_key_id=None, aws_secret_access_key=None,
                  **kwargs):
         # only import boto when needed to allow top-lvl s3 module import
         import boto
+        import boto.s3.connection
         from boto.s3.key import Key
 
         options = self._get_s3_config()
         options.update(kwargs)
         # Removing key args would break backwards compability
-        if not aws_access_key_id:
-            aws_access_key_id = options.get('aws_access_key_id')
-        if not aws_secret_access_key:
-            aws_secret_access_key = options.get('aws_secret_access_key')
-        for key in ['aws_access_key_id', 'aws_secret_access_key']:
+        role_arn = options.get('aws_role_arn')
+        role_session_name = options.get('aws_role_session_name')
```

Comment:
If I don't include `aws_role_arn` and/or `aws_role_session_name` within my config, am I correct that line 86 and 87 will throw errors (something to the effect of 'variable doesn't exist in luigi.cfg')?